build-ps/build-binary.sh

```diff
@@ -423,28 +432,63 @@ fi
     fi
 
     # NORMAL TARBALL
-    cd "$INSTALLDIR/usr/local/$PRODUCT_FULL"
-    link
+    if [[ ${WITH_ZENFS} != "ON" ]]; then
+        cd "$INSTALLDIR/usr/local/$PRODUCT_FULL"
+        link
+    fi
 
     # MIN TARBALL
     if [[ $CMAKE_BUILD_TYPE != "Debug" ]]; then
         cd "$INSTALLDIR/usr/local/minimal/$PRODUCT_FULL-minimal"
         rm -rf mysql-test 2> /dev/null
         find . -type f -exec file '{}' \; | grep ': ELF ' | cut -d':' -f1 | xargs strip --strip-unneeded
-        link
+        if [[ ${WITH_ZENFS} != "ON" ]]; then
+            link
+        fi
     fi
 )
 
 # Package the archive
 (
+    function build_and_place_zenfs_utils {
+        if [[ -f $INSTALLDIR/zenfs ]]; then
+            echo "ZenFS utils is built"
+        else
+            SOURCEDIR=${WORKDIR_ABS}/rocksdb-source
+            INSTALL_ROOT=${WORKDIR_ABS}/rocksdb-root
+            BUILD_ROOT=${WORKDIR_ABS}/rocksdb-build
+
+            mkdir ${WORKDIR_ABS}/rocksdb-root ${WORKDIR_ABS}/rocksdb-build
+
+            pushd $SOURCEDIR
+            CC=clang-12 CXX=clang++-12 make DISABLE_WARNING_AS_ERROR=1 PREFIX=${INSTALL_ROOT}/usr OBJ_DIR=${BUILD_ROOT} ROCKSDB_PLUGINS=zenfs -j$(nproc) install-static
+            popd
+
+            pushd $SOURCEDIR/plugin/zenfs/util
+            PKG_CONFIG_PATH=$INSTALL_ROOT/usr/lib/pkgconfig make CC=clang-12 CXX=clang++-12 -j$(nproc)
+            popd
+
+            cp $SOURCEDIR/plugin/zenfs/util/zenfs $INSTALLDIR/
+            rm -rf $INSTALL_ROOT $BUILD_ROOT $SOURCEDIR
+        fi
+    }
+
     cd "$INSTALLDIR/usr/local/"
     #PS-4854 Percona Server for MySQL tarball without AGPLv3 dependency/license
     find $PRODUCT_FULL -type f -name 'COPYING.AGPLv3' -delete
+    if [[ ${WITH_ZENFS} == "ON" ]]; then
+        build_and_place_zenfs_utils
```

Comment:
shouldn't we just call this function once and then only copy zenfs bin file here and on line 490?
It should be called on the build stage not on packaging stage